integrations/java/openhouse-java-runtime/src/main/java/com/linkedin/openhouse/javaclient/exception/WebClientRequestWithMessageException.java

```diff
@@ -17,4 +18,9 @@ public WebClientRequestWithMessageException(WebClientRequestException requestExc
   public String getMessage() {
     return requestException.getMessage();
   }
+
+  @Override
+  public int getStatusCode() {
+    return HttpStatus.BAD_REQUEST.value();
```

Comment:
Status code doesn't make sense for request, does it? Shouldn't this be set on only `WebClientResponseWithMessageException`, not in the base `WebClientWithMessageException`?

Reply:
We expect that handlers can use generic `WebClientWithMessageException` to catch both types, hence added the `getStatusCode`.

Reply:
hmm, to my understanding trino will be using `WebClientWithMessageException` to catch both `Request,Response` exceptions. since `getStatusCode` is only relevant to `WebClientResponse`, I wasn't sure how to reflect this if `WebClientWithMessageException` caught a `Request`. if we add it only to `WebClientResponseWithMessageException`, wouldn't the user have to explicitly catch the `Response` exception in order to use the `getStatusCode`?

Reply:
> If we add it only to WebClientResponseWithMessageException, wouldn't the user have to explicitly catch the Response exception in order to use the getStatusCode?

Yeah this is expected.

If you don't care about the status code, and just want to catch exceptions, you would have this:
```java
try {
  openhouse.foo();
} catch (WebClientWithMessageException e) {
  log.error("failed!", e);
}
```

But if you care about the status code, this doesn't really make sense:
```java
try {
  openhouse.foo();
} catch (WebClientWithMessageException e) {
  // getStatusCode() doesn't make sense if it's a _request_ exception so this is kind of wrong
  log.error("failed with status code {}", e.getStatusCode(), e);
}
```
Rather you should have either this:
```java
try {
  openhouse.foo();
} catch (WebClientResponseWithMessageException e) {
  log.error("response failed with status code {}", e.getStatusCode(), e);
} catch (WebClientRequestWithMessageException e) {
  log.error("other error", e);
}
```
Or, what would probably be better:
```java
try {
  openhouse.foo();
} catch (WebClientResponseWithMessageException e) {
  log.error("response failed with status code {}", e.getStatusCode(), e);
} catch (WebClientWithMessageException e) {
  log.error("other error", e);
}
```
This is just like how you might, at any time, catch more specific exceptions first, then more generic ones, e.g.:
```java
try {
  // some code
} catch (IllegalArgumentException e) {
  log.error("bad argument", e);
} catch (RuntimeException e) {
  log.error("other error", e);
}
```

Reply:
that makes sense. let me move `getStatusCode` into `Response` only